open_academy/models/open_academy_session.py

```diff
@@ -1,14 +1,54 @@
-from odoo import models, fields
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
 
 
 class OpenAcademySession(models.Model):
     _name = "open.academy.session"
     _description = "model of session on open academy"
 
     name = fields.Char(required=True, size=50)
-    duration = fields.Text()
-    start_date = fields.Datetime()
-    number_seats = fields.Integer()
+    duration = fields.Integer()
+    active = fields.Boolean(default=True)
+    start_date = fields.Datetime(default=fields.Date.context_today)
+    end_date = fields.Datetime()
+    number_seats = fields.Integer(required=True)
+    number_seats_taken = fields.Float(compute="_compute_number_taken_seats")
     instructor_id = fields.Many2one("res.partner", "instructor")
     course_id = fields.Many2one("open.academy.course", required=True)
     attendees_ids = fields.Many2many("res.partner", "session_res_partner_rel")
+    attendees_count = fields.Integer(compute="_compute_number_taken_seats", store=True)
+
+
+    @api.depends("number_seats", "attendees_ids")
+    def _compute_number_taken_seats(self):
+        for record in self:
+            number_attenndees = len(record.attendees_ids)
+            record.attendees_count = number_attenndees
+            if number_attenndees == 0 or record.number_seats == 0:
+                record.number_seats_taken = 0
+                return
```

Comment:
But normally we don't use the return statement in computed methods, only if it is necessary in special cases. And this is not a special case. If you add an else statement, you can avoid the return please

```suggestion
                record.number_seats_taken = 0
           else:
                record.number_seats_taken = (number_attenndees / record.number_seats) * 100
```